### PR TITLE
Add support for cells with non-local storage

### PR DESCRIPTION
**Pull request by Jezurko**
- [x] Move storage attributes from `hl` to `core`
- [x] Add Interface for storage information on declarations
- [x] Modify cell declarations to keep storage information
- [x] Nuke GlobalRefs and replace them with DeclRefs
- [x] Lower non-local cells to llvm 
- [x] Add tests

### CODE REVIEW

File: test/vast/Dialect/HighLevel/glob-linkage-a.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// CHECK: hl.var @wundef, <weak>
+int __attribute__((weak)) wundef;
+// CHECK: hl.var @wdef, <weak>
+int __attribute__((weak)) wdef = 5;
+// CHECK: hl.var @ewdef, <weak>
+extern int __attribute__((weak)) ewdef = 5;
+// CHECK: hl.var @edef, <external>
+extern int edef = 5;
+// CHECK: hl.var @undef, <common>
+int undef;
```
Comment (Jezurko) [inline test/vast/Dialect/HighLevel/glob-linkage-a.c:14]: From my understanding this is correct (see [clang/llvm](https://llvm.org/docs/LangRef.html#linkage-types) documentation), but when testing this in [godbolt](https://godbolt.org/z/KzKTKzre8) or locally it actually has `external` linkage. Not sure how to proceed with this

File: lib/vast/Conversion/FromHL/Passes.cpp [hunk 2]
```diff
@@ -42,6 +42,10 @@ namespace vast::conv::pipeline {
         return pass(createVarsToCellsPass);
     }
 
+    pipeline_step_ptr evict_static_locals() {
+        return pass(createEvictStaticLocalsPass);
```
Comment (xlauko) [inline lib/vast/Conversion/FromHL/Passes.cpp:46]: add `.depends_on(refs_to_ssa)`?

File: test/vast/Conversion/subscript-b.c [hunk 3]
```diff
@@ -2,25 +2,25 @@
...
+// STD_TYPES: hl.var @arr1, <external> : !hl.lvalue<!hl.array<3, si32>> = {
 
 
-// VAL_CAT: hl.var @arr1 : !hl.ptr<!hl.array<3, si32>> = {
+// VAL_CAT: hl.var @arr1, <external> : !hl.ptr<!hl.array<3, si32>> = {
 // VAL_CAT:    hl.value.yield {{.*}} : !hl.array<3, si32>
 
 
-// C_LLVM:  llvm.mlir.global internal constant @arr1() {addr_space = 0 : i32} : !llvm.array<3 x i32> {
+// C_LLVM:  llvm.mlir.global external @arr1() {addr_space = 0 : i32} : !llvm.array<3 x i32> {
```
Comment (xlauko) [inline test/vast/Conversion/subscript-b.c:12]: <3

**Review by xlauko — APPROVED** [review]

File: lib/vast/Conversion/FromHL/Passes.cpp (hunk 2 above)
Comment (Jezurko) [reply lib/vast/Conversion/FromHL/Passes.cpp:46]: I don't think it depends on `refs_to_ssa`. I believe these passes can be run in any order, as the legalizers should make sure that the pass does not act on the wrong Variable declarations

**Pull request merged**